internal/nets/nets.go

```diff
@@ -87,25 +70,55 @@ func List(offset, limit int, mls paramstores.NetParamStore) ([]types.BriefNet, i
 	return nets, cursor, nil
 }
 
-// Required returns the number of patterns required to train a net with the specified topology
-func Required(inputs, outputs, hLayers int, conf config.Config) (int, error) {
-	switch conf.ML.Net {
-	case "mlp":
-		topology := MLPTopology(inputs, outputs, hLayers)
-		w := 0
-		for layer, neurons := range topology[:len(topology)-1] {
-			w += (neurons + 1) * topology[layer+1]
+// LoadNetwork is used to retrieve network params for a given ID and build a network object with them. When they can't
+// be found (nil, nil) will be returned
+func LoadNetwork(id, nType string, nps paramstores.NetParamStore) (Network, error) {
+	switch nType {
+	case types.MultilayerPerceptron:
+		var np paramstores.MLPParams
+		found, err := nps.Load(id, &np)
+		if err != nil {
+			return nil, err
+		}
+		if !found {
+			return nil, nil
 		}
-		return int(math.Ceil(float64(w) / 0.1)), nil
+		return MLPFromParams(id, np)
 	default:
-		return 0, errors.New(conf.ML.Net + " is not a valid net type")
+		return nil, errors.New(nType + " is not a valid net type")
 	}
 }
 
+// Required returns the number of patterns required to train a net (of the most demanding type) with the specified
+// topology
+func Required(inputs, outputs, hLayers int, conf config.Config) int {
+	// required := []int{}
+
+	// MLP
+	topology := MLPTopology(inputs, outputs, hLayers)
+	w := 0
+	for layer, neurons := range topology[:len(topology)-1] {
+		w += (neurons + 1) * topology[layer+1]
+	}
+	/*
+		required = append(required, int(math.Ceil(float64(w)/0.1)))
+
+		max := 0
+		for index := range required {
+			if required[index] > max {
+				max = required[index]
+			}
+		}
+		return max
+	*/
+
+	return int(math.Ceil(float64(w) / 0.1))
+}
+
 // Trainer listens for requests to train neural nets with new points
 func Trainer(c chan types.TrainRequest, conf config.Config) error {
 	// Set up ml store
-	mls, err := paramstores.New(conf)
+	nps, err := paramstores.New(conf)
 	if err != nil {
 		logger.Error("Failed to initialize ml store for the training service", err)
```

Comment:
In this message, "ml store" is correct?

Reply:
I missed those two, thanks! Changed and updated